react/components/MinicartIconButton.tsx

```diff
@@ -11,16 +11,32 @@ const CSS_HANDLES = ['minicartIconContainer', 'minicartQuantityBadge'] as const
 
 interface MinicartIconButtonProps {
   quantityDisplay: MinicartIconButtonType
+  itemCountMode: MinicartTotalItemsType
+}
+
+interface ProductWithQuantity {
+  quantity: number
 }
 
 const MinicartIconButton: FC<MinicartIconButtonProps> = ({
   quantityDisplay,
+  itemCountMode,
 }) => {
   const { orderForm, loading }: OrderFormContext = useOrderForm()
   const handles = useCssHandles(CSS_HANDLES)
   const { open, openBehavior, openOnHoverProp } = useMinicartState()
   const dispatch = useMinicartDispatch()
-  const itemQuantity = loading ? 0 : orderForm.items.length
+  const totalItemsSum = (arr: ProductWithQuantity[]) =>
+    arr.reduce(
+      (sum: number, product: ProductWithQuantity) => sum + product.quantity,
+      0
+    )
```

Comment:
The `orderForm` returned from `useOrderFormContext` is correctly typed, so the type you want to use for the arguments of this function is of type `OrderFormItem` (as defined at https://github.com/vtex-apps/minicart/blob/master/react/typings/OrderForm.ts#L6).

Also, please move the function definition outside of the component so that it doesn't get recreated in every render.